src/libraries/System.Linq/tests/ContainsTests.cs

```diff
@@ -26,7 +26,7 @@ public void SameResultsRepeatCallsStringQuery()
                     where !string.IsNullOrEmpty(x)
                     select x;
 
-            Assert.Equal(q.Contains("X"), q.Contains("X"));
+            Assert.Equal(q.Contains('X'), q.Contains('X'));
```

Comment:
What exactly does this test? the two arguments of Assert.Equal are exactly the same, how it may fail?

Reply:
Just noticed that this change isn't correct because q isn't string. I'm reverting it now.

However, I still have the same question, if possible to answer.

```suggestion
            Assert.Equal(q.Contains("X"), q.Contains("X"));
```

Reply:
It's validating idempotency of the operation, e.g. that the first call doesn't somehow erroneously mutate the source such that the second call gets a different answer.